### PR TITLE
HW 6 JS Arrays

### PR DESCRIPTION
**Pull request by limur88**

### CODE REVIEW

File: Arrays-6/task1.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+function getEmployeeinfo(name){
+    const employees = ["Jason", "Hanna", "Bill", "Allice", "Kate"];
+    const wages = [100, 125, 95, 70, 88];
+    let answer = [name];
+    if (employees.indexOf(name) == -1){
+        return "null"
+    } else{
+    let index = employees.indexOf(name);
+    //let answer = [name, wages[index]];
+    answer.push(wages[index]);
```
Comment (anatoly-karpovich) [inline Arrays-6/task1.js]:
```suggestion
    const index = employees.indexOf(name)
    if (index == -1){
        return "null"
    } else{
    const answer = [name,  wages[index]];
```

File: Arrays-6/task2.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+function comparePizzas (my, other){
+    let unique = [...my];
+    for(let i = 0; i< other.length; i++){
+           if (my.indexOf(other[i]) != -1) {
+            unique.splice(unique.indexOf(other[i]), 1);
+        }
+    } if (unique.length == 0){
+        return "Null";
+    }return unique
```
Comment (anatoly-karpovich) [inline Arrays-6/task2.js]:
```suggestion
    } 
return unique.length ? unique : null
```

File: Arrays-6/task2.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
+// 1. У вас есть массив названий пицц вашего конкурента. Создайте функцию, которая будет принимать ваш набор названий пицц (массив) 
+//   и возвращать только те, которых нет у конкурента (тоже массив). Если все ваши пиццы есть у конкурента - вернуть null
+//   Пиццы конкурента:
+//   const competitorPizzas = ['Peperoni', 'Caprichosa', 'Diablo', '4 cheeses', 'hawai'] 
+const competitorPizzas = ['Peperoni', 'Caprichosa', 'Diablo', '4 cheeses', 'hawai'];
+// const myPizzas = ['Peperoni', 'Caprichosa', 'Diablo', '4 cheeses', 'hawai'];
+const myPizzas = ['Peperoni','4 meat', 'Belladonna', 'Caprichosa', 'Diablo', '4 cheeses'];
+
+function comparePizzas (my, other){
+    let unique = [...my];
```
Comment (anatoly-karpovich) [inline Arrays-6/task2.js]: Checks for that task should be case insensitive

File: Arrays-6/task2.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+console.log(comparePizzas (myPizzas, competitorPizzas))
+
+// 2. Написать функцию, которая принимает предложение (слова разделенные только пробелами) в качестве параметра и выводит в консоль слово с наибольшим количеством букв. 
+//   Если таких слов несколько - выводит их все.
+
+const sentence = "I love to read book"
+  function returnLongestWord(sentence){
+    let words = sentence.split(" ");
+    let wordLength = [];
+    for (let word of words){
```
Comment (anatoly-karpovich) [inline Arrays-6/task2.js]:
```suggestion
    for (const word of words){
```

File: Arrays-6/task2.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        wordLength.push(word.length);
+    }
+    
+    let greatest = Math.max(...wordLength);
+    let answer = []
+    for (let word of words){
```
Comment (anatoly-karpovich) [inline Arrays-6/task2.js]:
```suggestion
    for (const word of words){
```

File: Arrays-6/task2.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        if (word.length == 4){
```
Comment (anatoly-karpovich) [inline Arrays-6/task2.js]:
```suggestion
        if (word.length == greatest){
```

File: Arrays-6/task2.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    while (word[i] == word[word.length - 1 -i] && i< word.length){
+    count ++;
+    i++;
+}
+if (count == word.length){
+    let answer = "Palyndrom";
+    return answer
+}else{
+    let answer = "NOT palyndrom";
+    return answer
```
Comment (anatoly-karpovich) [inline Arrays-6/task2.js]:
```suggestion
return count == word.length
```

File: Arrays-6/task2.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+}
+}
+console.log(checkIfPolyndrom ('Anna'))
```
Comment (anatoly-karpovich) [inline Arrays-6/task2.js]:
```suggestion
console.log(`Word is palindrom: ${checkIfPolyndrom ('Anna')}`)
```

File: Arrays-6/task3.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
+// 5*. Напишите рекурсивную функцию, которая принимает на вход число 
+//и складывает его цифры. 
+//   Если сумма получилась больше 9 - снова сложите цифры. 
+//И так пока, сумма не станет меньше либо равной 9. 
+//   После окончания сложений возвращает полученное число.
+// Например при подаче числа 19 (1+9=10>9, потому 1+0=1) 
+//выводится 1
+
+function sumOfDigits(num){
+    let digits = num.toString().split("");
```
Comment (anatoly-karpovich) [inline Arrays-6/task3.js]:
```suggestion
    const digits = num.toString().split("");
```

**Review by anatoly-karpovich — CHANGES_REQUESTED** [review]

**Review by VladimirLQA — DISMISSED** [review]

File: Arrays-6/task2.js [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+let numbers = [1, 1, 1, 1, 5, 5, 10, 3, 8, 10, 1, 1, 1, 5, 5, 6, 9];
+for(let i = 0; i < numbers.length -1; i++){
+    for (let j = i + 1; j < numbers.length; j++){
+        if(numbers[i] === numbers[j]){
+            numbers.splice(j, 1);
+            j--;
+        }
+    }
+}
```
Comment (anatoly-karpovich) [inline Arrays-6/task2.js]:
```suggestion
const result = []
for(const num of numbers){
if(! result.includes(num)) result.push(num)
}
```

File: Arrays-6/task2.js [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+        answer.push(word);
+    }
+}
+    return answer;
+  }
+  console.log(returnLongestWord("I love to read book"))
+
+// 3. Напишите функцию, которая принимает на вход массив чисел, убирает из него дубликаты и возвращает массив с только уникальными значениями.
+
+let numbers = [1, 1, 1, 1, 5, 5, 10, 3, 8, 10, 1, 1, 1, 5, 5, 6, 9];
```
Comment (anatoly-karpovich) [inline Arrays-6/task2.js]: There is "function" word in a task, please follow requirements)

**Review by anatoly-karpovich — DISMISSED** [review]

**Review by anatoly-karpovich — APPROVED** [review]

File: Arrays-6/task3.js [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+console.log(sumOfDigits(189));
+
+// 6*. Написать функцию, которая принимает на вход строку с текстом,
+// и заменяет каждую пару стоящих подряд идентичных букв на одну 
+//следующую в алфавите, 
+//     и так пока в тексте не останется двух одинаковых букв 
+//стоящих рядом (через пробел и другой знак препинания можно)
+//     Пример: aabc => bbc => cc => d
+
+function cutDoubles(str){
```
Comment (VladimirLQA) [inline Arrays-6/task3.js:29]: Try with this case `zzabc`

File: Arrays-6/task3.js [hunk 13]
```diff
@@ -0,0 +1,40 @@
+// 5*. Напишите рекурсивную функцию, которая принимает на вход число 
+//и складывает его цифры. 
+//   Если сумма получилась больше 9 - снова сложите цифры. 
+//И так пока, сумма не станет меньше либо равной 9. 
+//   После окончания сложений возвращает полученное число.
+// Например при подаче числа 19 (1+9=10>9, потому 1+0=1) 
+//выводится 1
+
+function sumOfDigits(num){
```
Comment (VladimirLQA) [inline Arrays-6/task3.js:9]: Try  to add handling for negative numbers, for example, `-19`

**Review by VladimirLQA — APPROVED** [review]

**Pull request merged**